util/src/main/java/io/kubernetes/client/informer/cache/Controller.java

```diff
@@ -113,7 +129,7 @@ public void run() {
 
     synchronized (this) {
       // TODO(yue9944882): proper naming for reflector
-      reflector = new ReflectorRunnable<ApiType, ApiListType>(apiTypeClass, listerWatcher, queue);
+      reflector = newReflector();
```

Comment:
@brendandburns Thought about injecting a ReflectorFactory to improve testability. But then I chose this way so we do not add another class and we still can choose this option later.